rust/core-lib/src/config.rs

```diff
@@ -254,6 +254,21 @@ impl ConfigManager {
         }
     }
 
+    /// Copies the contents of `client_example.toml` to `preferences.xiconfig`. 
+    /// Used when an existing config file is not found.
+    pub fn create_example_preference_config(&self) -> Result<Option<PathBuf>, ConfigError> {
+        let client_example_config = include_str!("../assets/client_example.toml");
+        
+        if let Some(config_dir) = self.config_dir.as_ref() {
+            let config_file = config_dir.join("preferences.xiconfig");
+            let mut file = fs::File::create(config_file.as_path())?;
+            file.write_all(client_example_config.as_bytes())?;
```

Comment:
I'd _very_ much prefer "atomically" (well, -ish) writing the file here by writing it to a temp dir first and then moving it to the actual location

Reply:
See https://github.com/xi-editor/xi-editor/blob/4107776ddc82340eadb30b53ccbcb3435fd7ba61/rust/core-lib/src/file.rs#L202